container-host-files/opt/scf/bin/klog.sh

```diff
@@ -5,192 +5,82 @@ set -e
-        kubectl describe pods "${POD}" --namespace "${NS}" > "${DIR}/describe-pod.txt"
-    done
+		# Get the CF logs inside the pod if there are any
+		if [ "$(get_phase)" != 'Succeeded' ]; then
+			if [ check_for_log_dir ]; then
+				kubectl cp -n "${NS}" "${POD}":/var/vcap/sys/log/ "${POD_DIR}/" 2>/dev/null
+			fi
+		fi
 
-    # Unzip any logrotated files so the lookup can read them
-    gunzip -r "${KLOG}"
+		# Retrieving all the containers within a pod
+		CONTAINERS=($(kubectl get pods "${POD}" -n cf -o jsonpath='{.spec.containers[*].name}'))
 
-    kubectl get all --export=true --namespace "${NS}" -o yaml > "${KLOG}/${NS}/resources.yaml"
-    kubectl get events --export=true --namespace "${NS}" -o yaml > "${KLOG}/${NS}/events.yaml"
+		# Iterate over containers and dump logs
+		for CONTAINER in "${CONTAINERS[@]}"; do
 
-    touch "${DONE}"
-fi
-
-NEWLINE=0
-function lookfor {
-    read PATTERN
-
-    cd "${KLOG}"
-    if grep -c -r -F "${PATTERN}" > .grep; then
-        [ "${NEWLINE}" == "1" ] && echo
-        NEWLINE=1
-
-        echo ">>> ${PATTERN}"
-        echo
-        grep -v :0$ .grep
-
-        while read INFO; do
-            echo "${INFO}"
-        done
-    fi
-}
-
-# any pod with consul agent /var/vcap/sys/log/consul_agent/consul_agent.stdout.log
-lookfor <<EOF
-[ERR] agent: failed to sync remote state: No known Consul servers
-
-This happened on a consul agent that came up before the consul server,
-then abandoned the server as non-functional and then had no servers to
-talk with, restarting the pod fixed the issue (though restarting consul
-would also fix it).
-EOF
-
-read -d '' MSG <<EOF || true
-SST has happened twice and should only ever happen once. Probably missing the
-IST patch. If that patch is in place then full recovery of MySQL required:
+			CON_DIR="${POD_DIR}/${CONTAINER}"
+			mkdir -p ${CON_DIR}
 
-monit stop all # Check for lingering processes and kill them
-rm -rf /var/vcap/store/mysql/
-/var/vcap/jobs/mysql/bin/pre-start
-monit start all
-EOF
-
-# mysql pod /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-[ERROR] WSREP: SST failed: 1 (Operation not permitted)
-
-${MSG}
-EOF
-
-# mysql pod /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-[Warning] InnoDB: Cannot open table mysql/gtid_slave_pos from the internal data dictionary of InnoDB
-
-${MSG}
-EOF
-
-# mysql pod /var/vcap/sys/log/mysql/mysql.err.log
-lookfor <<EOF
-[Warning] WSREP: no nodes coming from prim view, prim not possible
-
-Nodes all shut down improperly, MySQL must be manually rebootstrapped. Will fail
-if the IST patch has not been applied.
-EOF
+			# Get the pod logs - previous may not be there if it was successful on the first run.
+			# Unfortunately we can't get anything past the previous one
+			kubectl logs "${POD}" -n "${NS}" -c ${CONTAINER} >"${CON_DIR}/kube.log"
+			kubectl logs "${POD}" -n "${NS}" -c ${CONTAINER} >"${CON_DIR}/kube-previous.log" 2>/dev/null || true
```

Comment:
This appears to be missing a `--previous`

Reply:
Added it back, thanks for catching this.